src/app/hostname-generators/views/HostnameGeneratorListView.vue

```diff
@@ -0,0 +1,126 @@
+<template>
+  <RouteView
+    name="hostname-generator-list-view"
+    :params="{
+      name: '',
+      page: 1,
+      size: 15,
+    }"
+    v-slot="{ route, t, can, uri, me }"
+  >
+    <AppView
+      :docs="t('hostname-generators.href.docs')"
+    >
+      <template #title>
+        <h1>
+          <RouteTitle
+            :title="t('hostname-generators.routes.items.title')"
+          />
+        </h1>
+      </template>
+      <div class="stack">
+        <div v-html="t('hostname-generators.routes.items.intro', {}, { defaultMessage: '' })" />
+        <KCard>
+          <DataLoader
+            :src="uri(sources, '/hostnamegenerators', {}, {
+              page: route.params.page,
+              size: route.params.size,
+            })"
+          >
+            <template
+              #loadable="{ data }"
+            >
+              <DataCollection
+                type="hostname-generators"
+                :items="data?.items ?? [undefined]"
+                :page="route.params.page"
+                :page-size="route.params.size"
+                :total="data?.total"
+                @change="route.update"
+              >
+                <AppCollection
+                  data-testid="hostname-generator-collection"
+                  :headers="[
+                    { ...me.get('headers.name'), label: t('hostname-generators.common.name'), key: 'name' },
+                    { ...me.get('headers.namespace'), label: t('hostname-generators.common.namespace'), key: 'namespace' },
+                    ...(can('use zones') ? [{ ...me.get('headers.zone'), label: t('hostname-generators.common.zone'), key: 'zone' }] : []),
+                    { ...me.get('headers.actions'), label: t('hostname-generators.common.actions'), key: 'actions', hideLabel: true },
+                  ]"
+                  :items="data?.items"
+                  :is-selected-row="(item) => item.name === route.params.name"
+                  @resize="me.set"
+                >
+                  <template #name="{ row: item }">
+                    <TextWithCopyButton
+                      :text="item.name"
+                    >
+                      <XAction
+                        data-action
+                        :to="{
+                          name: 'hostname-generator-summary-view',
+                          params: {
+                            name: item.id,
+                          },
+                          query: {
+                            page: route.params.page,
+                            size: route.params.size,
+                          },
+                        }"
+                      >
+                        {{ item.name }}
+                      </XAction>
+                    </TextWithCopyButton>
```

Comment:
Not necessary in the slightest, but if we need to push another commit or so onto this PR we could switch this to use `XCopyButton` while we are here.

Reply:
Ah I remember 💡 thanks 👍